raster/r.out.gdal/attr.c

```diff
@@ -47,12 +49,149 @@ int export_attr(GDALDatasetH hMEMDS, int band, const char *name,
      * GFU_MaxCount
      */
 
-    /* TODO: cats.ncats > 0 && rcount > 0
-     * how to merge categories and color rules ?
-     * what to do for a cell value that has a category but no color rule ?
-     * what to do for a cell value that has a color rule but no category ?
-     */
-    if (cats.ncats > 0) {
+    if (cats.ncats > 0 && rcount > 0) {
+        int use_minmax = 0;
+        int r1, g1, b1, r2, g2, b2;
+
+        /* merge categories and color rules:
+         * go through categories and fetch corresponding color
+         * write out value, label, red, green, blue
+         * or the minmax variant
+         */
+
+        if (maptype == CELL_TYPE) {
+            for (i = 0; i < cats.ncats; i++) {
+                label = Rast_get_ith_c_cat(&cats, i, &CellMin, &CellMax);
```

Comment:
This looks very good to me, and a great addition!

I have only minor suggestions: (1) `label` which is set on L64 and L73 (and L199 and L208) are never used: the return value may just as well be ignored on these lines; (2) `use_minmax` would be better represented by a bool.

Reply:
(1) done and (2) done. OK?

Reply:
I understand, but with `(void)Rast_get_ith_c_cat()` you just trick the format checker to not complain about unused return codes. Therefore I think my proposed solution where the code now also checks if the call to `Rast_get_ith_c_cat()` was successful (there is someting in label) is the formally correct solution.

Reply:
(This came up with `scan-build`, as "unused assignment" or something similar. Occasionally, I run it on new code or substantial updates).

I didn't check for possible erroring of `Rast_get_ith_c_cat()`, so with your update there's actually a code improvement (not only for better "code reading").